app/services/session_encryptor.rb

```diff
@@ -1,5 +1,14 @@
 class SessionEncryptor
-  @user_access_key = nil
+  def self.build_user_access_key
+    env = Figaro.env
+    UserAccessKey.new(env.session_encryption_key, env.password_pepper)
+  end
+
+  @user_access_key = build_user_access_key
+
+  class << self
+    attr_reader :user_access_key
```

Comment:
I'd probably prefer the class method to this `attr_reader` -- while I understand what you're doing here, I think a class method is more readable